scripts/test_db_connection_pool_load.sh

```
@@ -98,16 +101,57 @@ monitor_resources() {
+                cpu=0
+            fi
+        fi
         cpu_samples+=("$cpu")
 
-        # Memory usage
-        local mem=$(free | grep Mem | awk '{printf "%.2f", $3/$2 * 100.0}')
+        # Memory usage - portable detection
+        local mem
+        if [[ "$OSTYPE" == "darwin"* ]]; then
+            # macOS - calculate memory usage percentage
+            local total_mem=$(sysctl -n hw.memsize)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)